virtool/groups/data.py

```diff
@@ -150,6 +151,20 @@ async def delete(self, group_id: int):
             mongo_session,
             pg_session,
         ):
+            # Check if the group is associated with any users
+            user_associations = await pg_session.execute(
+                select(user_group_associations).where(
+                    user_group_associations.c.group_id == group_id
+                )
+            )
+
+            if user_associations:
+                # If the group is associated with any users, remove the associations
+                await pg_session.execute(
+                    delete(user_group_associations).where(
+                        user_group_associations.c.group_id == group_id
+                    )
+                )
```

Comment:
Again, this can probably be super-simplified using `CASCADE`.

Generally, Postgres/SQL is a lot more features to make our lives easier when compared with Mongo. I don't know them all. If you find yourself manually joining or filtering data in code or writing a lot of logic like this block, check if there are better ways to do it using features of Postgres.

On top of that, SQLAlchemy may have helpful functionality for making it easier to use these features of Postgres.